src/main/java/com/faforever/client/chat/AbstractChatTabController.java

```diff
@@ -420,6 +421,7 @@ private void sendMessage() {
       messageTextField.setDisable(false);
       messageTextField.requestFocus();
     }).exceptionally(throwable -> {
+      throwable = throwable instanceof CompletionException ? throwable.getCause() : throwable;
```

Comment:
I believe this could be used in other services as well. Maybe make it a method in some Util class? If we have one?
In this class it's already used twice.

Reply:
Would it really be worth it to extract a single, very simple ternary operator? The assignment will stay anyway and you will need to jump into a util code just to read this simple line of code.

Reply:
Well, from all the stacktraces I have seen so far, we should always unwrap the CompletionException and only occasionally do it, so I think a function called `unwrapCompletionException` would be used in many places and is easier to read than the ternary operator. But that is just my opinion.

Reply:
k, i guess `ConcurrentUtil` i a good class for that method.